### ISSUE TITLE
Rule Request: `require-presentational-children`

### ISSUE THREAD

**Issue by MelSumner** [ember-template-lint/ember-template-lint#1578]
There are roles that require all children to be presentational. 

While browsers are required to automatically apply `role="presentation"` to all descendant elements of any element with a role that cannot support semantic children, it would be useful to have this in a linting rule, so that the developer can receive this feedback in their developer workflow instead of after an accessibility audit has occurred or a user has filed a bug. 

Supporting document: https://w3c.github.io/aria-practices/#children_presentational

Good: 
```
<li role="tab">Tab title</li>
```

Bad:
```
<li role="tab"><h3>Tab Title</h3></li>
```

**Comment by faith-or** [comment]
Here's my attempt at this: https://github.com/ember-template-lint/ember-template-lint/pull/1982 

If @MelSumner  can give me some feedback. Not sure if I have the full list of non-semantic children.
